collect_app/src/main/java/org/odk/collect/android/configure/SettingsUtils.java

```diff
@@ -0,0 +1,29 @@
+package org.odk.collect.android.configure;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+
+import org.odk.collect.android.formmanagement.FormUpdateMode;
+import org.odk.collect.android.preferences.GeneralKeys;
+import org.odk.collect.android.preferences.Protocol;
+
+public class SettingsUtils {
+
+    private SettingsUtils() {
+
+    }
+
+    @NonNull
+    public static FormUpdateMode getFormUpdateMode(Context context, SharedPreferences generalSharedPreferences) {
```

Comment:
I was incredibly tempted to add a Settings interface for abstracting away this kind of logic (and shared prefs themselves). This would be in a weird place however as in most places it's pretty simple to just grab the actual pref value to work out what "state" we're in. I think if these utils start to grow we can revisit the idea.